plutus-ir/src/Language/PlutusIR.hs

```diff
@@ -155,7 +155,7 @@ data Term tyname name a =
                         | Error a (Type tyname a)
                         | IWrap a (Type tyname a) (Type tyname a) (Term tyname name a)
                         | Unwrap a (Term tyname name a)
-                        deriving (Functor, Show, Eq, Generic)
+                        deriving (Functor, Show, Generic)
```

Comment:
@michaelpj I just dropped the `Eq` instances for IR data types, because those were direct syntactic equality checks. Do we need the `Eq` instances? If so, should I add them in this PR? I'd prefer to do that later.

Reply:
Ah, yes, `deriving Eq` is broken now for things that have `Type`s and `Term`s in them. We now have to use top-level `deriving instance blah-blah-blah`.

Reply:
I don't think we *should* need those instances, unless we're doing `==` on errors with them in or something?

Reply:
We do things like `errOrX == Right x` and that requires `Eq` instances for errors. And there are some actual `==` on errors:

```haskell
    fmt "{- program " @?= Left (LexErr "Error in nested comment at line 1, column 12")
```

I do not want to refactor those, so I'd keep the `Eq` instances.